platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_tasks.adoc

```diff
@@ -289,5 +289,60 @@ include::{snippetsPath}/bestPractices/groupTasks-do/tests/groupTasks.good.out[]
 - <<writing_tasks.adoc#task_group_and_description,Task Group and Description>>
 
 === Tags
-
 `<<tags_reference.adoc#tag:tasks,#tasks>>`
+
+[[avoid_eager_file_collection_apis]]
+== Avoid using eager APIs on File Collections
+
+When working with Gradle's file collection types, be careful to avoid triggering dependency resolution during the configuration phase.
```

Comment:
Does Gradle have a way to assert that this was not done? Otherwise, it feels like a nice FR for a strict mode